src/functions.php

```diff
@@ -194,3 +194,17 @@ function is_string_array($input) {
     return true;
 }
 
+function recursive_copy($element) {
```

Comment:
Let's preface this with a doc block, and mark it `@internal` to advise users not to call it directly.

```
/**
 * Performs a deep copy of a value.
 *
 * This function will clone objects and recursively copy values within arrays.
 *
 * @internal
 * @see https://bugs.php.net/bug.php?id=49664
 * @param mixed $element Value to be copied
 * @return mixed
 */
```

Reply:
Related reading: http://phpsadness.com/sad/50